### PR TITLE
Replace Windows compatibility with WSL

### PR DESCRIPTION
**Pull request by ygoumaz**
Currently, the documentation and tutorials detail how to use `webots_ros2` on Windows and the package officially supports installation on this platform. However, using ROS2 outside of a Linux environment is not trivial and many other distributed packages are not maintained for macOS or Windows.

This PR deprecates the compatibility with Windows, but adds the possibility to run the package in WSL (Windows Subsystem for Linux). This way Webots still runs on Windows, while the whole ROS2 part can be run in a Linux environment, which greatly increases the simplicity of package maintenance, usage and the possibility to use other distributed packages. In short, the experience of Windows users is greatly improved and simplified.

In addition, this PR changes the way Webots is searched in the system at launch. The order of priority is as follows:
- Installation contained in ROS2_WEBOTS_HOME, regardless of version.
- Installation contained in WEBOTS_HOME, regardless of version.
- Search in the default installation folders (compatible version only):
  - `C:\Program Files\Webots` on the WSL system
  - `/usr/local/webots` and `/snap/webots/current/usr/share/webots` on Linux
  - `/Applications/Webots.app` on macOS
- Suggest an automatic installation of the compatible version of Webots in `~/.ros/webots`.

Wiki and documentation updates:
* [Getting Started](https://github.com/cyberbotics/webots_ros2/wiki/Getting-Started).
* [Complete Installation Guide](https://github.com/cyberbotics/webots_ros2/wiki/Complete-Installation-Guide).
* [ROS2 documentation](https://github.com/cyberbotics/ros2_documentation/pull/2).

### CODE REVIEW

File: webots_ros2_driver/webots_ros2_driver/urdf_spawner.py [hunk 1]
```diff
@@ -28,6 +30,9 @@ def get_webots_driver_node(event, driver_node):
 
 class URDFSpawner(ExecuteProcess):
     def __init__(self, output='log', name=None, urdf_path=None, robot_description=None, relative_path_prefix=None, translation='0 0 0', rotation='0 0 1 0', normal=False, box_collision=False, init_pos=None, **kwargs):
+        if is_wsl() and relative_path_prefix:
+            relative_path_prefix = subprocess.check_output(['wslpath', '-w', relative_path_prefix]).strip().decode("utf-8").replace('\\', '/')
```
Comment (omichel) [inline webots_ros2_driver/webots_ros2_driver/urdf_spawner.py:34]:
```suggestion
            relative_path_prefix = subprocess.check_output(['wslpath', '-w', relative_path_prefix]).strip().decode('utf-8').replace('\\', '/')
```

File: webots_ros2_mavic/launch/robot_launch.py [hunk 2]
```diff
@@ -37,13 +38,17 @@ def generate_launch_description():
         world=PathJoinSubstitution([package_dir, 'worlds', world])
     )
 
+    controller_url = ''
+    if is_wsl():
+        controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/'
```
Comment (omichel) [inline webots_ros2_mavic/launch/robot_launch.py:43]:
```suggestion
    controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/' if is_wsl() else ''
```

File: webots_ros2_epuck/launch/robot_launch.py [hunk 3]
```diff
@@ -39,6 +40,10 @@ def generate_launch_description():
         world=PathJoinSubstitution([package_dir, 'worlds', world])
     )
 
+    controller_url = ''
+    if is_wsl():
+        controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/'
```
Comment (omichel) [inline webots_ros2_epuck/launch/robot_launch.py:45]:
```suggestion
    controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/' if is_wsl() else ''
```

File: webots_ros2_driver/webots_ros2_driver/webots_launcher.py [hunk 4]
```diff
@@ -148,12 +160,16 @@ def _shutdown_process(self, context, *, send_sigint):
 
 class Ros2SupervisorLauncher(Node):
     def __init__(self, output='screen', respawn=True, **kwargs):
+        controller_url = ''
+        if is_wsl():
+            controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/'
```
Comment (omichel) [inline webots_ros2_driver/webots_ros2_driver/webots_launcher.py:165]:
```suggestion
        controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/' if is_wsl() else ''
```

File: webots_ros2_tesla/launch/robot_launch.py [hunk 5]
```diff
@@ -37,13 +38,17 @@ def generate_launch_description():
         world=PathJoinSubstitution([package_dir, 'worlds', world])
     )
 
+    controller_url = ''
+    if is_wsl():
+        controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/'
```
Comment (omichel) [inline webots_ros2_tesla/launch/robot_launch.py:43]:
```suggestion
    controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/' if is_wsl() else ''
```

File: webots_ros2_tests/test/test_system_driver.py [hunk 6]
```diff
@@ -50,19 +51,23 @@ def generate_test_description():
         mode='fast'
     )
 
+    controller_url = ''
+    if is_wsl():
+        controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/'
```
Comment (omichel) [inline webots_ros2_tests/test/test_system_driver.py:56]:
```suggestion
    controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/' if is_wsl() else ''
```

File: webots_ros2_tiago/launch/robot_launch.py [hunk 7]
```diff
@@ -48,6 +49,10 @@ def generate_launch_description():
         mode=mode
     )
 
+    controller_url = ''
+    if is_wsl():
+        controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/'
```
Comment (omichel) [inline webots_ros2_tiago/launch/robot_launch.py:54]:
```suggestion
    controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/' if is_wsl() else ''
```

File: webots_ros2_turtlebot/launch/robot_launch.py [hunk 8]
```diff
@@ -39,6 +40,10 @@ def generate_launch_description():
         world=PathJoinSubstitution([package_dir, 'worlds', world])
     )
 
+    controller_url = ''
+    if is_wsl():
+        controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/'
```
Comment (omichel) [inline webots_ros2_turtlebot/launch/robot_launch.py:45]: Same comment here.

File: webots_ros2_universal_robot/launch/multirobot_launch.py [hunk 9]
```diff
@@ -42,6 +43,10 @@ def get_ros2_nodes(*args):
     ur5e_control_params = os.path.join(package_dir, 'resource', 'ros2_control_config.yaml')
     abb_control_params = os.path.join(package_dir, 'resource', 'ros2_control_abb_config.yaml')
 
+    controller_url = ''
+    if is_wsl():
+        controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/'
```
Comment (omichel) [inline webots_ros2_universal_robot/launch/multirobot_launch.py:48]: Same comment here.

File: webots_ros2_universal_robot/launch/robot_launch/robot_nodes_launch.py [hunk 10]
```diff
@@ -34,6 +35,10 @@ def generate_launch_description():
     robot_description = pathlib.Path(ur5e_urdf_path).read_text()
     ros2_control_params = os.path.join(package_dir, 'resource', 'ros2_control_config.yaml')
 
+    controller_url = ''
+    if is_wsl():
+        controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/'
```
Comment (omichel) [inline webots_ros2_universal_robot/launch/robot_launch/robot_nodes_launch.py:40]:
```suggestion
    controller_url = 'tcp://' + get_wsl_ip_address() + ':1234/' if is_wsl() else ''
```

**Review by omichel — CHANGES_REQUESTED** [review]
It looks good to me (although I didn't test it yet).

**Review by omichel — APPROVED** [review]
Thank you.
As discussed, the wiki page should be improved.

**Comment by ygoumaz** [comment]
I updated the 2 following pages:
- https://github.com/cyberbotics/webots_ros2/wiki/Getting-Started
- https://github.com/cyberbotics/webots_ros2/wiki/Complete-Installation-Guide

Let me know if this is ok for you.

**Comment by omichel** [comment]
As discussed, we should remove the macOS instructions from the wiki as we won't support macOS directly, but only through docker.

**Comment by ygoumaz** [comment]
The documentation about macOS is removed for now, until the docker solution is ready.

**Pull request merged**